packages/client/src/errors.ts

```diff
@@ -87,6 +90,36 @@ export function isGrpcServiceError(err: unknown): err is GrpcServiceError {
   );
 }
 
+/**
+ * Returns true if the provided error or its cause is a {@link GrpcServiceError} with code DEADLINE_EXCEEDED.
+ *
+ * @see {@link Connection.withDeadline}
+ */
+export function isGrpcDeadlineError(err: unknown): err is Error {
+  while (isError(err)) {
+    if (isGrpcServiceError(err) && (err as GrpcServiceError).code === status.DEADLINE_EXCEEDED) {
+      return true;
+    }
+    err = (err as any).cause;
+  }
```

Comment:
There is no error type that includes the "cause" field? Is that the reason you cast to `any`?

Reply:
> There is no error type that includes the "cause" field?

That's it. Or to be very exact, there is no _generic_ error that includes it. The `cause` field in JS is a wide-spread convention, but the TS standard lib definition doesn't formally include it. Now, I could cast the error to `TemporalFailure` just to get the field, but I think in this case that would be even worse than casting to `any`.